### PR TITLE
Add root tree node uuid in calls to study server

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/redux/actions.js [hunk 1]
```diff
@@ -304,3 +304,12 @@ export function selectFavoriteContingencyLists(favoriteContingencyLists) {
         [PARAM_FAVORITE_CONTINGENCY_LISTS]: favoriteContingencyLists,
     };
 }
+
+export const SELECTED_TREE_NODE = 'SELECTED_TREE_NODE';
+
+export function selectTreeNode(selectTreeNode) {
+    return {
+        type: SELECTED_TREE_NODE,
+        selectTreeNode: selectTreeNode,
```
Comment (niconoir) [inline src/redux/actions.js:313]: Rename as selectedTreeNode ?

File: src/components/line-menu.js [hunk 2]
```diff
@@ -54,6 +54,7 @@ const withLineMenu =
         const intlRef = useIntlRef();
 
         const studyUuid = decodeURIComponent(useParams().studyUuid);
+        const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
Comment (niconoir) [inline src/components/line-menu.js:57]: Maybe pass selectedNodeUuid as a prop if possible ?

**Review by niconoir — CHANGES_REQUESTED** [review]
Some comments related to redux usage

File: src/redux/reducer.js [hunk 3]
```diff
@@ -89,6 +90,7 @@ const paramsInitialState = {
 
 const initialState = {
     studyUuid: null,
+    selectTreeNode: null,
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/redux/reducer.js]: Done

File: src/components/dialogs/connectivity-edition.js [hunk 4]
```diff
@@ -63,6 +64,8 @@ const ConnectivityEdition = ({
 }) => {
     const classes = useStyles();
     const studyUuid = decodeURIComponent(useParams().studyUuid);
+    const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/connectivity-edition.js]: Done

File: src/components/contingency-list-selector.js [hunk 5]
```diff
@@ -52,6 +52,8 @@ const ContingencyListSelector = (props) => {
         (state) => state[PARAM_FAVORITE_CONTINGENCY_LISTS]
     );
 
+    const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/contingency-list-selector.js]: Done

File: src/components/dialogs/equipment-deletion-dialog.js [hunk 6]
```diff
@@ -55,6 +56,7 @@ const useStyles = makeStyles(() => ({
  */
 const EquipmentDeletionDialog = ({ open, onClose }) => {
     const studyUuid = decodeURIComponent(useParams().studyUuid);
+    const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/equipment-deletion-dialog.js]: Done

File: src/components/dialogs/generator-creation-dialog.js [hunk 7]
```diff
@@ -50,6 +51,7 @@ const GeneratorCreationDialog = ({ open, onClose, voltageLevelOptions }) => {
     const classes = useStyles();
 
     const studyUuid = decodeURIComponent(useParams().studyUuid);
+    const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/generator-creation-dialog.js]: Done

File: src/components/dialogs/line-creation-dialog.js [hunk 8]
```diff
@@ -55,6 +56,7 @@ const LineCreationDialog = ({ open, onClose, voltageLevelOptions }) => {
     const classes = useStyles();
 
     const studyUuid = decodeURIComponent(useParams().studyUuid);
+    const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/line-creation-dialog.js]: Done

File: src/components/dialogs/load-creation-dialog.js [hunk 9]
```diff
@@ -48,6 +49,7 @@ const LoadCreationDialog = ({ open, onClose, voltageLevelOptions }) => {
     const classes = useStyles();
 
     const studyUuid = decodeURIComponent(useParams().studyUuid);
+    const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/load-creation-dialog.js]: Done

File: src/components/dialogs/two-windings-transformer-creation-dialog.js [hunk 10]
```diff
@@ -58,6 +59,7 @@ const TwoWindingsTransformerCreationDialog = ({
     const classes = useStyles();
 
     const studyUuid = decodeURIComponent(useParams().studyUuid);
+    const selectedNodeUuid = useSelector((state) => state.selectTreeNode);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/dialogs/two-windings-transformer-creation-dialog.js]: Done

File: src/redux/actions.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/redux/actions.js]: Done

File: src/components/line-menu.js (hunk 2 above)
Comment (FranckLecuyer) [reply src/components/line-menu.js]: Done

**Review by niconoir — APPROVED** [review]

File: src/redux/reducer.js [hunk 11]
```diff
@@ -89,6 +90,7 @@ const paramsInitialState = {
 
 const initialState = {
     studyUuid: null,
+    selectedTreeNode: null,
```
Comment (SlimaneAmar) [inline src/redux/reducer.js:93]: Use an object study {uuid, nodeuuid } ?

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**